cpp/src/arrow/compute/compute-test.cc

```diff
@@ -343,6 +402,16 @@ TEST_F(TestCast, FloatingPointToInt) {
                                                     options);
 }
 
+TEST_F(TestCast, IntToFloatingPoint) {
+  auto options = CastOptions::Safe();
+
+  vector<bool> is_valid = {true, false, true, true, true};
```

Comment:
is_valid not used. Since you added a null check in the functor, I presume you meant to use this for a test?